File: abc379/e/tle.py
import sys

sys.set_int_max_str_digits(0)
sys.setrecursionlimit(10**6)


def main():
    N = int(input())
    S = input()
    ans = 0
    prev = 0
    for i in range(N):
        d = ord(S[i]) - ord('0')
        now = prev * 10 + d * (i + 1)
        prev = now
        ans += now
    print(ans)


if __name__ == "__main__":
    main()

File: abc379/e/tle.cc
#include <algorithm>
#include <cassert>
#include <iostream>
#include <vector>
#include <boost/multiprecision/cpp_int.hpp>
using namespace std;
using namespace boost;

int main() {
  int N;
  cin >> N;
  string S;
  cin >> S;
  multiprecision::cpp_int ans = 0;
  multiprecision::cpp_int prev = 0;
  for (int i = 0; i < N; i++) {
    int d = S.at(i) - '0';
    multiprecision::cpp_int now = prev * 10;
    now += d * (i + 1);
    prev = now;
    ans += now;
  }
  cout << ans << endl;
  return 0;
}
